webapp/src/Controller/API/ProblemController.php

```diff
@@ -102,9 +102,12 @@ public function addProblemsAction(Request $request): array
         // Note: we read the JSON as YAML, since any JSON is also YAML and this allows us
         // to import files with YAML inside them that match the JSON format
         $data = Yaml::parseFile($file->getRealPath(), Yaml::PARSE_DATETIME);
-        if ($this->importExportService->importProblemsData($contest, $data, $ids)) {
+        if ($this->importExportService->importProblemsData($contest, $data, $ids, $messages)) {
             return $ids;
         }
+        if (!empty($messages)) {
+            throw new BadRequestHttpException($this->dj->jsonEncode($messages));
+        }
         throw new BadRequestHttpException("Error while adding problems");
```

Comment:
Wouldn't it be better to merge the contents of these exceptions into a single one? So that it is both clear where the error happens and what the underlying cause is (if any messages are there). I think this would also make the logic of the code here less weird: first conditionally throw an exception, and then finally still throw an exception.

Reply:
So you suggest:

```php
$message = "Error while adding problems";
if (!empty($messages)) {
	$message .= ': ' . $this->dj->jsonEncode($messages);
}
throw new BadRequestHttpException($message);
```
?

Reply:
Yes, that looks good to me!